hooks/shotgun_get_published_file.py

```diff
@@ -9,75 +9,31 @@
 
 """
 import sgtk
-from sgtk import TankError
 
 HookBaseClass = sgtk.get_hook_baseclass()
 
 
 class GetPublishedFile(HookBaseClass):
-    _PUBLISHED_FILE_FIELDS = ["path", "task", "entity"]
 
-    def execute(self, entity_type, entity_id, published_file_entity_type, **kwargs):
-        """
-        Given a Version, PublishedFile or TankPublishedFile,
-        return the PublishedFile or TankPublishedFile linked to it.
-
-        For a Version with multiple PublishedFiles,
-        return the first PublishedFile matching
-        `viewer_extensions` configuration parameter.
-        If no PublishedFile matches the extensions,
-        return the first one in the list.
-
-        :param str entity_type: Version, PublishedFile or TankPublishedFile.
-        :param int entity_id: The id of the entity.
-        :param str published_file_entity_type: PublishedFile or TankPublishedFile.
-        :returns: The selected PublishedFile
-        :raises: TankError if a PublishedFile cannot be found.
-        """
-        if entity_type == "Version":
-            # entity is a Version so try to get the id
-            # of the published file it is linked to
-            if published_file_entity_type == "PublishedFile":
-                published_files_field = "published_files"
-            else:
-                published_files_field = "tank_published_file"
-
-            v = self.parent.shotgun.find_one("Version", [["id", "is", entity_id]], [published_files_field])
-            if not v.get(published_files_field):
-                raise TankError("Sorry, this can only be used on Versions with an associated published file.")
-            if len(v[published_files_field]) == 1:
-                return self.resolve_single_file(
-                    published_file_entity_type,
-                    v[published_files_field][0],
-                )
-            else:
-                return self.resolve_multiple_files(
-                    published_file_entity_type,
-                    v[published_files_field]
-                )
-        else:
-            # entity is PublishedFile or TankPublishedFile. Return it
-            return self.published_file(entity_type, entity_id)
-
-    def resolve_single_file(self, entity_type, published_file):
+    def resolve_single_file(self, published_file_type, published_file):
         """
         Decide wether or not to return the published file, or raise a TankError.
         This default implementation returns it.
 
-        :param str entity_type: PublishedFile or TankPublishedFile.
+        :param str published_file_type: PublishedFile or TankPublishedFile.
         :param dict published_file: The published file.
         :returns: The published file with the right fields.
         """
-        return self.published_file(entity_type, published_file["id"])
+        return self.parent.published_file(published_file_type, published_file["id"])
```

Comment:
This not right: you shouldn't use custom methods from the parent, only standard TK methods.

Reply:
You could have a hook base class somewhere in your app and use it with `create_hook_instance` if you need to share base code between hooks? https://developer.shotgunsoftware.com/tk-core/platform.html#sgtk.platform.Engine.create_hook_instance
Or simply duplicate the method...